086-monitoring-of-custom-resources.md

```diff
@@ -0,0 +1,61 @@
+# Monitoring of custom resources
+
+In https://github.com/strimzi/strimzi-kafka-operator/issues/9802 there was a discussion about how to monitor the state of a custom resources (CR) like `KafkaTopic`, `KafkaUser`, ... .
+For a simple and Kubernetes native monitoring, [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) (KSM) can be used which utilizes e.g. the status object of a Kubernetes resource.
+
+## Current situation
+
+Strimzi provides metrics for several use cases but currently not for the state of `KafkaTopic` and `KafkaUser` CRs.
+TODO need input for the status of other CRs and their metrics, e.g. `Kafka`, `StrimziPodSet`, `KafkaRebalance`...
```

Comment:
The metric name is `strimzi_resource_state`. It is currently issued only by the Cluster Operator and only for the user level resources (e.g. Kafka, KafkaConnect ... but not for StrimziPodSet resources). If you wanna include some examples here:

```
$ curl -s localhost:8080/metrics | grep strimzi_resource_state
# HELP strimzi_resource_state Current state of the resource: 1 ready, 0 fail
# TYPE strimzi_resource_state gauge
strimzi_resource_state{kind="Kafka",name="my-cluster",reason="none",resource_namespace="myproject",} 1.0
strimzi_resource_state{kind="KafkaConnect",name="my-connect",reason="none",resource_namespace="myproject",} 1.0
```

Or as a screenshot:
![Screenshot 2024-09-23 at 16 19 56](https://github.com/user-attachments/assets/d08ad287-f1cf-4012-a5d3-ca612a4d6c18)